warehouse/email/__init__.py

```diff
@@ -33,12 +35,25 @@ def _compute_recipient(user, email):
 
 
 @tasks.task(bind=True, ignore_result=True, acks_late=True)
-def send_email(task, request, recipient, msg):
+def send_email(task, request, recipient, msg, user_id, ip_address):
     msg = EmailMessage(**msg)
     sender = request.find_service(IEmailSender)
+    from_address = parseaddr(sender.from_address)[1]
 
     try:
         sender.send(recipient, msg)
+
+        user_service = request.find_service(IUserService, context=None)
+        user_service.record_event(
+            user_id,
+            tag="account:email:sent",
+            ip_address=ip_address,
```

Comment:
```suggestion
            ip_address=request.remote_addr,
```

Reply:
It seems that the `request` object in the `send_email` task is missing many of the attributes like `remote_addr` and `db`. Any ideas on why that could be?

Reply:
Are they getting lost on serialization? I.e. do they exist before it's passed to `delay`?

Reply:
Yes, it seems that the `WarehouseTask` class is overwriting the `request` that is passed into the task

Reply:
I have updated the function so that we now pass the `remote_addr` to the task along with a `redact_ip` flag for the UI mark as "Redacted"